packages/wrangler/templates/constellation-beta-facade.js

```diff
@@ -0,0 +1,326 @@
+// src/shim.ts
+import worker, * as OTHER_EXPORTS from "__ENTRY_POINT__";
+export * from "__ENTRY_POINT__";
+import { Buffer } from "node:buffer";
```

Comment:
Does this shim require `node:buffer`? What happens if a user doesn't have `nodejs_compat` flag enabled? Sorry if I missed any validation around this - not 100% familiar with wrangler's codebase but couldn't spot it at a glance.

Reply:
We do use it currently. I guess we can mention it in the documentation of the project.

Reply:
If I can't be removed, I'd definitely encourage this to be validated in wrangler to prevent the use of `constellation` bindings without this flag. Otherwise people are going to be hit with pretty obscure errors if they try and use it without this flag.

For example when I don't have this flag enabled: ![](https://up.jross.me/uz23pemb)

A new user is going to be pretty overwhelmed with that error stack (why is it loading `node:buffer`? I didn't tell it to to that? etc.) vs validation before it gets this far such as "Constellation currently requires the `nodejs_compat` compatibility flag to be enabled in your worker. Add the following to your `wrangler.toml`..."